RealmSwift/Realm.swift

```diff
@@ -366,6 +370,80 @@ import Realm.Private
         return rlmRealm.inWriteTransaction
     }
 
+#if REALM_ASYNC_WRITES
+// MARK: Asynchronous Transactions
+
+    /**
+     Commits transaction block asynchronously.
+     - parameter `block` will be added to the ssynchronous transaction queue.
+     - parameter `onComplete` will be called after commit has reached stable storage.
+     Write  blocks for the multiple calls will be executed in order.
+     */
+    @discardableResult
+    public func writeAsync(_ block: @escaping (AsyncTransactionId) -> Void, _ onComplete: (() -> Void)? = nil) throws -> AsyncTransactionId {
+        try beginAsyncWrite { asyncTransactionId in
+            block(asyncTransactionId)
+            commitAsyncWrite(onComplete)
+        }
+    }
+
+    /**
+     Begins asynchronous write transaction.
+     - parameter `block` The block containing actions to perform.
+     `block` should end by calling `commitAsyncWrite`, `cancelAsyncWrite`,
+     `commitWrite` or `cancelWrite`.
+     Returning without one of these calls will be equivalent to calling `commitAsyncWrite`.
+     Return Asynchronous transaction's handle.
+     @note `block` is queued for execution on the scheduler associated with the current realm.
+     It will run after the write mutex has been acquired.
+     The call returns immediately allowing the caller to proceed while the write mutex is held by someone else.
+     Write blocks from multiple calls to `beginAsyncWrite` or `writeAsync`
+     will be executed in order.
+     A later call to `beginAsyncWrite` or `writeAsync` will wait for any earlier
+     write blocks.
+     */
+    @discardableResult
+    public func beginAsyncWrite(_ asyncWriteBlock: @escaping (AsyncTransactionId) -> Void) throws -> AsyncTransactionId {
+        var asyncTransactionId: AsyncTransactionId = 0
+        asyncTransactionId = rlmRealm.beginAsyncWriteTransaction {
+            asyncWriteBlock(asyncTransactionId)
+        }
+        return asyncTransactionId
+    }
+
+    /** Commit asynchronous transaction.
+     - parameter onComplete  is queued for execution on the scheduler associated with
+     the current realm. It will run after the commit has reached stable storage.
+     - parameter isGroupingAllowed     If `true`, the next `commitAsyncWrite` *may* run without an
+     intervening synchronization of stable storage.  Such a sequence of commits form a group.
+     It may help to have a better performance on write.
```

Comment:
I recommend changing `It` to `isGroupingAllowed`. So it's clearer that isGroupingAllowed may be performant.

Reply:
Nice